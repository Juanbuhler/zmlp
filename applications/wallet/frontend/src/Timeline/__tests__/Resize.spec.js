import TestRenderer, { act } from 'react-test-renderer'

import TimelineResize from '../Resize'

const noop = () => () => {}

describe('<TimelineResize />', () => {
  it('should render properly', () => {
    const component = TestRenderer.create(
      <TimelineResize
        dispatch={noop}
        zoom={100}
        videoRef={{
          current: { duration: 18, currentTime: 0 },
        }}
        rulerRef={{
          current: { scrollLeft: 0, scrollWidth: 100 },
        }}
      />,
    )

    act(() => {})

    // Zoom in to 200%
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom In' }).props.onClick()
    })

    // Zoom out to 100%
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    // Attempt to zoom out further than minimum zoom of 100%
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    expect(
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.isDisabled,
    ).toBe(true)

    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should render properly when ref values are null', () => {
    const component = TestRenderer.create(
      <TimelineResize
        dispatch={noop}
        zoom={100}
        videoRef={{ current: null }}
        rulerRef={{ current: null }}
      />,
    )

    act(() => {})

    // Zoom in to 200%
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom In' }).props.onClick()
    })

    // Zoom out to 100%
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    // Attempt to zoom out further than minimum zoom of 100%
    act(() => {
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.onClick()
    })

    expect(
      component.root.findByProps({ 'aria-label': 'Zoom Out' }).props.isDisabled,
    ).toBe(true)

    expect(component.toJSON()).toMatchSnapshot()
  })
})
