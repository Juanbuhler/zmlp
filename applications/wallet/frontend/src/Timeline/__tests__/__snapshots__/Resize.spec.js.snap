// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimelineResize /> should render properly 1`] = `
.emotion-0 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #4a4a4a;
  border-radius: 2px;
  background-color: #252525;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  z-index: 30;
  opacity: 0.8;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 0;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  opacity: 1;
}

.emotion-1:hover {
  color: #73b61c;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-2 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 0;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  opacity: 1;
}

.emotion-2:hover {
  color: #73b61c;
}

.emotion-2[aria-disabled=true] {
  color: #808080;
}

<div
  className="emotion-0"
>
  <button
    aria-disabled={true}
    aria-label="Zoom Out"
    className="emotion-1"
    onClick={[Function]}
    type="button"
  >
    <svg
      data-file-name="circleMinus.svg"
      height={20}
    />
  </button>
  <button
    aria-label="Zoom In"
    className="emotion-2"
    onClick={[Function]}
    type="button"
  >
    <svg
      data-file-name="circlePlus.svg"
      height={20}
    />
  </button>
</div>
`;

exports[`<TimelineResize /> should render properly when ref values are null 1`] = `
.emotion-0 {
  position: absolute;
  bottom: 16px;
  right: 16px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border: 1px solid #4a4a4a;
  border-radius: 2px;
  background-color: #252525;
  box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.5);
  z-index: 30;
  opacity: 0.8;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 8px;
  padding-right: 8px;
}

.emotion-1 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 0;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: not-allowed;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  opacity: 1;
}

.emotion-1:hover {
  color: #73b61c;
}

.emotion-1[aria-disabled=true] {
  color: #808080;
}

.emotion-2 {
  -webkit-flex: 0;
  -ms-flex: 0;
  flex: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  border-radius: 2px;
  padding: 0;
  white-space: nowrap;
  font-size: 15px;
  line-height: 19px;
  font-weight: 500;
  border: none;
  cursor: pointer;
  color: #ffffff;
  background-color: transparent;
  padding: 8px;
  opacity: 1;
}

.emotion-2:hover {
  color: #73b61c;
}

.emotion-2[aria-disabled=true] {
  color: #808080;
}

<div
  className="emotion-0"
>
  <button
    aria-disabled={true}
    aria-label="Zoom Out"
    className="emotion-1"
    onClick={[Function]}
    type="button"
  >
    <svg
      data-file-name="circleMinus.svg"
      height={20}
    />
  </button>
  <button
    aria-label="Zoom In"
    className="emotion-2"
    onClick={[Function]}
    type="button"
  >
    <svg
      data-file-name="circlePlus.svg"
      height={20}
    />
  </button>
</div>
`;
